dtfs-central-api/src/v1/controllers/portal/deal/add-deal-comment.controller.js

```diff
@@ -1,54 +1,75 @@
+const { validateAuditDetails, generateAuditDatabaseRecordFromAuditDetails } = require('@ukef/dtfs2-common/change-stream');
+const { InvalidAuditDetailsError } = require('@ukef/dtfs2-common/errors');
 const { MONGO_DB_COLLECTIONS } = require('@ukef/dtfs2-common');
 const { ObjectId } = require('mongodb');
 const db = require('../../../../drivers/db-client').default;
 const { findOneDeal } = require('./get-deal.controller');
 
-const addDealComment = async (_id, commentType, comment) => {
+const addDealComment = async (_id, commentType, comment, auditDetails) => {
   const collection = await db.getCollection(MONGO_DB_COLLECTIONS.DEALS);
 
-  if (ObjectId.isValid(_id)) {
-    const findAndUpdateResponse = await collection.findOneAndUpdate(
-      { _id: { $eq: ObjectId(_id) } },
-      {
-        $push: {
-          [commentType]: {
-            $each: [
-              {
-                ...comment,
-                timestamp: Date.now(),
-              },
-            ],
-            $position: 0,
-          },
+  if (!ObjectId.isValid(_id)) {
+    return { status: 400, message: 'Invalid Deal Id' };
+  }
+
+  const auditRecord = generateAuditDatabaseRecordFromAuditDetails(auditDetails);
+
+  const findAndUpdateResponse = await collection.findOneAndUpdate(
+    { _id: { $eq: ObjectId(_id) } },
+    {
+      $push: {
+        [commentType]: {
+          $each: [
+            {
+              ...comment,
+              auditRecord,
+              timestamp: Date.now(),
+            },
+          ],
+          $position: 0,
         },
       },
-      { returnNewDocument: true, returnDocument: 'after' },
-    );
+    },
```

Comment:
Think this is in the wrong place
```suggestion
    {
      $push: {
        [commentType]: {
          $each: [
            {
              ...comment,
              auditRecord,
              timestamp: Date.now(),
            },
          ],
          $position: 0,
        },
      },
      $set: {
        auditRecord,
      }
    },
```

Reply:
great spot